### PR TITLE
feat(test): add travis and appveyor

### LINKED ISSUE #2: set up CI
appveyor and travis

### PR DESCRIPTION
**Pull request by ashleygwilliams**
fixes #2

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        make install)
+      - export OPENSSL_DIR=$HOME/openssl-musl
+    script: cargo build --release --target $TARGET --locked
+
+  # dist OSX binary
+  - os: osx
+    rust: stable
+    env: MACOSX_DEPLOYMENT_TARGET=10.7 DEPLOY=1 TARGET=x86_64-apple-darwin
+    script: cargo build --release --target $TARGET --locked
+    install: true
```
Comment (killercup) [inline .travis.yml]: Not sure what these dist builds do aside from building the project right now? Are they still useful if we don't do deploys (cf #7)?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  # dist linux binary
+  - env: TARGET=x86_64-unknown-linux-musl DEPLOY=1
+    rust: stable
+    before_script:
+      - rustup target add $TARGET
+      - curl https://www.openssl.org/source/openssl-1.0.2l.tar.gz | tar xzf -
+      - (cd openssl-1.0.2l &&
+        CC=musl-gcc ./Configure --prefix=$HOME/openssl-musl no-dso no-ssl2 no-ssl3 linux-x86_64 -fPIC &&
+        make -j$(nproc) &&
+        make install)
```
Comment (killercup) [inline .travis.yml]: Is this truly necessary? Or is it actually cargo cult? ;) (In both cases it'd be great to add a comment so I don't as you again in 2 months)

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+language: rust
+sudo: false
+
+matrix:
+  include:
+
+  # tests pass
+  - rust: stable
+    script:
+    - cargo test --locked
```
Comment (killercup) [inline .travis.yml]: FYI I tend to add `--all` in case this ever becomes a workspace and I forget about updating this

File: .travis.yml (hunk 1 above)
Comment (ashleygwilliams) [reply .travis.yml]: oh yeah whoops. i can fix, i am just cargo culting from the wasm-pack one, which has lot more going on 😅

File: .travis.yml (hunk 2 above)
Comment (ashleygwilliams) [reply .travis.yml]: lol cargo cult :P will tear out in a sec

**Comment by ashleygwilliams** [comment]
so... lolsob, this is gonna fail cuz currently stable rust's `rustfmt` is borked:

```
The command "rustup component add rustfmt-preview" exited with 0.
0.10s$ cargo fmt --all -- --check
Unrecognized option: 'check'
```

i am assured this will go away by the next release this thursday so i'm gonna split the test into tests and style and let CI fail on just this check for now

**Comment by killercup** [comment]
maybe pin to beta in the meantime?

**Comment by ashleygwilliams** [comment]
@killercup eh i am fine with letting the style check fail for 2 days, i dont anticipate a ton of churn on this until then

**Pull request merged**